### ISSUE TITLE
no update ops in tf.GraphKeys.UPDATE_OPS collection

### ISSUE THREAD

**Issue by twangnh** [LevinJ/SSD_tensorflow_VOC#24]
Hi! Levin, by setting breakpoint, I found after constructing the network, `tf.get_collections(tf.GraphKeys.UPDATE_OPS)` returns empty list, in my experiences, that should already contain update ops of batchnorm layers,  but when inspect the trained checkpoint file, the moving average values of batchnorm params had actually been updated, this is really strange, could you give some idea on this issue? Thanks !

**Comment by LevinJ** [comment]
Hi @MrWanter , this behavior is what we expect from current code implementation.

The slim.batch_norm  API prototype,

```
batch_norm(inputs,
               decay=0.999,
               center=True,
               scale=False,
               epsilon=0.001,
               activation_fn=None,
               param_initializers=None,
               param_regularizers=None,
               updates_collections=ops.GraphKeys.UPDATE_OPS,
               is_training=True,
               reuse=None,
               variables_collections=None,
               outputs_collections=None,
               trainable=True,
               batch_weights=None,
               fused=None,
               data_format=DATA_FORMAT_NHWC,
               zero_debias_moving_mean=False,
               scope=None,
               renorm=False,
               renorm_clipping=None,
               renorm_decay=0.99,
               adjustment=None)
```

if updates_collections=ops.GraphKeys.UPDATE_OPS is left as this default, then indeed moving mean and moving variance update ops will be added to this collection, but in our current ssd implementation, this parameter is set as `None` as below in ssd.py file, so the moving mean and moving variance ops are not added to this  collection, and a control dependency are added  inside  batch_norm function to make sure the updates are computed in place.


```
with slim.arg_scope([slim.conv2d],
                            activation_fn=None):
                with slim.arg_scope([slim.batch_norm],
                            activation_fn=tf.nn.relu, is_training=is_training,updates_collections=None):
                    with slim.arg_scope([slim.dropout],
                            is_training=is_training,keep_prob=keep_prob):
                        with tf.variable_scope(self.model_name):
                            return self.__additional_ssd_block(end_points, net)
```

**Comment by twangnh** [comment]
I see, thanks for your response!

**Comment by twangnh** [comment]
I noticed that gamma of batchnorm is not used, as presented by the documentation:
```
    scale: If True, multiply by `gamma`. If False, `gamma` is
      not used. When the next layer is linear (also e.g. `nn.relu`), this can be
      disabled since the scaling can be done by the next layer.
```
would there be possible improvement for adding gamma term? have you tried this?

**Comment by Janezzliu** [comment]
@MrWanter Hi,I'm learning SSD too,would you mind learning together and discuss problems?This is my email:2107404138@qq.com
